go-controller/pkg/ovn/ovn.go

```diff
@@ -564,28 +629,36 @@ func (oc *Controller) WatchPods() {
 				}
 				return
 			}
+			needsRetry := false
 			if podScheduled(pod) {
 				if err := oc.addLogicalPort(pod); err != nil {
 					klog.Errorf(err.Error())
 					oc.recordPodEvent(err, pod)
-					retryPods.Store(pod.UID, true)
+					needsRetry = true
 				}
 			} else {
 				// Handle unscheduled pods later in UpdateFunc
-				retryPods.Store(pod.UID, true)
+				needsRetry = true
+			}
+			if needsRetry {
+				oc.retryPodsLock.Lock()
+				defer oc.retryPodsLock.Unlock()
+				oc.retryPods[pod.UID] = retryEntry{pod, time.Now()}
```

Comment:
maybe a wrapper for this since you have one for deletion anyway?

Reply:
yeah will do